File: Data Structures & Algorithms/subarray-sum-equals-k/submission-8.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int ans=0;
        int currsum=0;
        Map<Integer,Integer> mp1=new HashMap<>();
        mp1.put(0,1);
        for(int i:nums)
        {
            currsum+=i;
            int diff=currsum-k;
            ans+=mp1.getOrDefault(diff,0);
            mp1.put(currsum,mp1.getOrDefault(currsum,0)+1);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-8.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int n=s.size();
        int index=0;
        int ans=0;
        vector<vector<int>>dp(n,vector<int>(n,0));
        for(int i=n-1;i>=0;i--)
        {
            for(int j=i;j<n;j++)
            {
                if(s[i]==s[j] &&(j-i<=2 || dp[i+1][j-1]))
                {
                    dp[i][j]=true;
                    if(ans<j-i+1)
                    {
                        ans=j-i+1;
                        index=i;
                    }
                }
            }
        }
        return s.substr(index,ans);
    }
};
